Automattic/jetpack
# Classic Theme Helper: Copy Testimonials Custom Post Type code to Classic Theme Helper package

Fixes https://github.com/Automattic/vulcan/issues/593

## Proposed changes:

* Moved the `testimonial.php` content across to the Classic Theme Helper package.
* Nested classes within a function have been moved to their own files.
* Ensured phan was happy
* I have tested requiring the package to make sure that where possible no additional changes should be needed in the package before requiring, in the next PR.
  * As such, some additional changes were made along the lines of changes made when the portfolio CPT via the CTH package was being required: https://github.com/Automattic/jetpack/pull/39268 (predominantly related to the `maybe_register_cpt` function, as well as `activation_post_type_support`).
* Functions to initialize the `Jetpack_Testimonial` class were not moved over - this will happen when we are requiring via the package.

### Other information:


## Jetpack product discussion

 https://github.com/Automattic/vulcan/issues/593

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* The package isn't being used yet, and no build is happening, but checking the format in which code has been moved / class set up could be useful here. 
* If you do generally want to test the feature, the Testimonials can be activated via Jetpack > Settings > Writing. You can test adding a new testimonial item, viewing it, adding a link to the new item via the `[testimonial]` shortcode. A theme that includes jetpack-testimonial is Lodestar.
* If testing, make sure you test with the changes applied for both the Jetpack Mu WPcom and Jetpack (so on simple using both generated commands below). On Simple, toggling testimonials is done via Settings > Writing (not Jetpack > Settings > Writing)
